tutorials/mpi-fortran/code/reduce_std.f90

```diff
@@ -0,0 +1,89 @@
+module subs
+  implicit none
+contains
+  subroutine create_rand_nums(rand_nums, num_elements)
+    ! Creates an array of random numbers. Each number has a value from 0 - 1
+    integer, intent(in) :: num_elements
+    real, intent(out)   :: rand_nums(num_elements)
+
+    integer :: i
+
+    do i = 1, num_elements
+      rand_nums(i) = rand()
+    end do
+
+  end subroutine create_rand_nums
+end module subs
+
+program main
+  use mpi_f08
+  use iso_fortran_env, only: error_unit
+  use subs
+
+  implicit none
+
+  integer :: num_args
+  character(12) :: arg
+  integer :: num_elements_per_proc
+  integer :: world_size, world_rank
+  real :: r, local_sum, global_sum, mean, local_sq_diff, global_sq_diff, stddev
+  real, allocatable :: rand_nums(:)
+  integer :: i
+
+  num_args = command_argument_count()
+
+  if (num_args .ne. 1) then
+    write (error_unit, *) 'Usage: reduce_std num_elements_per_proc'
+    stop
+  end if
+
+  call get_command_argument(1, arg)
+  read (arg, *) num_elements_per_proc
```

Comment:
As per previous comment use `iostat` optional argument in `read()` function to checking for dodgy input.